src/utils/validatorUtils.ts

```diff
@@ -108,7 +108,7 @@ export function getParameterValidators(parameter: ts.ParameterDeclaration, param
   );
 }
 
-export function getPropertyValidators(property: ts.PropertyDeclaration): Tsoa.Validators | undefined {
```

Comment:
Please explain why undefined is now an acceptable response?

@HoldYourWaffle, as a heads up, I’m unlikely to approve a PR that does a refactor of the entire code base without any clarity on what bug it fixes or what feature it adds to tsoa. I’ll give you time to respond, but it’s  likely that I will close this PR. I would however approved a smaller PR if it:
* targets one small feature of tsoa
* increases the automated code coverage for that feature

Ultimately I like to see software grow and iterate in small cycles not in “Big Bang” refactors. I’m saying this because you said that you have a large amount of changes coming.

By the way, tsoa will soon have PR and issue templates that clarify these contribution guidelines.

Reply:
> Please explain why undefined is now an acceptable response?

I'm not sure what you mean by this. If I remember correctly the `undefined` return type was "unused" because it would **always** return at least `{}`, it's just a typing fix.

> I’m unlikely to approve a PR that does a refactor of the entire code base

Well it's good I'm not doing that then 😉 I'm just cleaning up the type resolver for typesafety enhancements/fixes and especially readability. I also added some notes on uncovered edge-cases I think I might've found or pieces of code that were confusing to me (so they can be better documented in the future).

> without any clarity on what bug it fixes or what feature it adds to tsoa

As I said before, it increases readability (a lot in my opinion) and makes the code more typesafe. Sorry for not mentioning those reasons earlier, I thought [this](https://github.com/lukeautry/tsoa/pull/445#issuecomment-524551649) (the initial motivation) and [this](https://github.com/lukeautry/tsoa/pull/445#issuecomment-524563203) comment had made that clear already.

> I’ll give you time to respond

This isn't meant in any negative sense, just out of honest curiosity: how long would you have waited for my response? You seem to be pretty quick on these sort of things.

<br>

> Ultimately I like to see software grow and iterate in small cycles not in “Big Bang” refactors. I’m saying this because you said that you have a large amount of changes coming.

This is definitely not a "big bang refactor", I'd hardly even call it a refactor. I'm also not sure where you got the idea that "large changes were coming", did I choose my wording poorly somewhere? If so please tell me, I'm just trying to learn.

I'm only doing two things right now:

1. Fixing some typing mistakes I noticed (I don't see a reason why this would ever be a bad thing)
2. Making the code more readable by adapting more modern approaches and conciser code. 

Especially the second point is important for ensuring this project stays maintainable and most importantly **approachable for new contributors**.
The current code is not very newcomer-friendly (at least in my experience) because it contains all kinds of unnecessary "bloat constructions" such as:

- [extra variable + type assertions] instead of type guards
- One-use variables that could perfectly be inlined (more code to read and interpret, stuff like "what is this variable for?")
- Unnecessary `as any` or even blatantly wrong assertions that effectively ruin typesafety

There are more of these but these three are the most common (and in my opinion the most annoying) ones. They're not just stylistic choices, they make the code objectively less readable (although you could argue about that for the second type).
I won't pretend I didn't make *some* stylistic changes, but that's mostly out of habit. I'll do a full passover once I'm done to remove any of these *pure stylistic* changes.

<br>

The changes you're seeing right now are just some little mistakes/improvements I found/made while exploring the codebase for the very first time. They are meant to make the code easier to understand (at least for an outsider), most often by reducing/removing these unnecessary bloat constructions.
As long as these changes don't:

- Cause bugs (which we have tests for *[I promise I'll make sure they run this time]*)
- Increase confusion (which can vary from person to person, feedback is always welcome)
- Decrease test coverage
- Undermine typesafety (which is the opposite of what I'm doing)
- Reduce maintainability (I'm trying to increase it)
- Make the code less accessible for newcomers (Also doing the opposite of this)
- Reduce flexibility for no reason
- Increase complexity
- Decrease documentation
- Significantly change the layout/architecture of the project (which is what I would call a "(big bang) refactor")

I don't see why that's a bad thing.

Reply:
My two cents: if you say the codebase is confusing, then it is confusing. A lot of this code is cobbled together from (sometimes not very well thought out) bug fixes, so I trust that this all comes from a good place.

I think as a point of discipline, the type fixes should be their own PR as they represent no change in runtime behavior and should be uncontroversial if they are, indeed, corrections.

Other stuff like code style is fine but we should probably open up a broader discussion about how we want to approach style conventions. I was pleased to see that we added prettier. I think we should probably just pick someone else's existing style guide and apply it universally; to the extent possible, I don't want us arguing over things that don't result in some benefit for the end-user.

As it stands, I don't think we have a great contributor guide, so that's something we can work together to produce. For a long time, it wasn't necessary with me as benevolent dictator, but now that we've got several folks getting really active, we need to establish some conventions.

Reply:
> My two cents: if you say the codebase is confusing, then it is confusing. A lot of this code is cobbled together from (sometimes not very well thought out) bug fixes, so I trust that this all comes from a good place.

Thank you for joining in on this. The code reads to me exactly as you described it, hacked together with sometimes not-100%-perfect solutions. The idea is there, and it's not extremely terrible (after all I did get through it) but there's definitely some room for improvement.
I'm honestly just trying to help improve this project, nothing more nothing less. I think it's an amazing concept and I hope that by making the code easier to understand it can be more accessible for future contributors (myself included).

> I think as a point of discipline, the type fixes should be their own PR as they represent no change in runtime behavior and should be uncontroversial if they are, indeed, corrections.

I thought about doing that too. If it doesn't create too much of a conflict hell I'll try to split out these parts of the PR.

> Other stuff like code style is fine but we should probably open up a broader discussion about how we want to approach style conventions.

Agreed. I think most of my changes of which you could possibly say that they're "stylistic" rest on the same set of "style choices". If we decide on which style we want, we can reject/approve a whole set of these changes at once. Again if it doesn't create too much of a commit hell I might be able to split-out all these "style choices" (if you really want to call it that way) into separate commits.

> I don't want us arguing over things that don't result in some benefit for the end-user.

I definitely agree, as long as there's still room for maintainability improvements such as these (referring to the typing changes/improvements).

Reply:
> I thought about doing that too. If it doesn't create too much of a conflict hell I'll try to split out these parts of the PR.

It shouldn't be too bad. When in doubt, split it out ✂️ 

> I definitely agree, as long as there's still room for maintainability improvements such as these (referring to the typing changes/improvements).

Yeah, my point here is that we should just have a style guide with these things documented so at code review time there isn't even a discussion to be had. I opened #449 to kick off a high level discussion of how we might iterate on this idea.

Reply:
I agree. I'll add some of my opinions to your issue shortly.